src/Features/VisualBasic/Portable/CodeFixes/GenerateType/GenerateTypeCodeFixProvider.vb

```diff
@@ -56,7 +57,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeFixes.GenerateType
         End Function
 
         Protected Overrides Function GetTargetNode(node As SyntaxNode) As SyntaxNode
-            Return (DirectCast(node, ExpressionSyntax)).GetRightmostName()
+            Dim qualified = TryCast(node.Parent, QualifiedNameSyntax)
+            If qualified IsNot Nothing Then
+                Return GetRightMostName(qualified)
+            End If
+
+            Dim memberAccess = TryCast(node.Parent, MemberAccessExpressionSyntax)
+            If memberAccess IsNot Nothing Then
+                Return GetRightMostName(memberAccess)
+            End If
+
+            Return DirectCast(node, ExpressionSyntax).GetRightmostName
```

Comment:
I did a bit of a double-take here. :smile: I think the ```GetRightMostName``` helper below should have a different name than the ```GetRightMostName``` extension method referenced above. Perhaps a name that indicates why this helper is different from the extension method and why the extension method still needs to exist?